### PR TITLE
solution

### PR DESCRIPTION
**Pull request by RomSmile**

### CODE REVIEW

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,84 @@
...
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
```
Comment (george-w-boo) [inline src/robotFactory.js:43]: не розумію, наіщо цей рядок (

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,84 @@
...
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (this.currentLoad === null && cargo.weight <= this.maxLoadWeight) {
```
Comment (george-w-boo) [inline src/robotFactory.js:73]: `this.currentLoad === null` можна скоротити до `!this.currentLoad`

**Review by george-w-boo — APPROVED** [review]
В принципі норм 👍

File: src/robotFactory.js [hunk 3]
```diff
@@ -1,10 +1,84 @@
...
+      };
+    } else {
+      this.coords = coords;
+    }
+  }
 
-class DeliveryDrone {}
+  getInfo() {
+    // eslint-disable-next-line max-len
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
```
Comment (VinnieJ-2k20) [inline src/robotFactory.js:21]: Instead of disabling eslint, it's better to find another solution.

File: src/robotFactory.js [hunk 4]
```diff
@@ -1,10 +1,84 @@
...
+
+    if (!coords.z) {
+      this.coords = {
+        x: 0,
+        y: 0,
+        z: 0,
+      };
+    } else {
+      this.coords = coords;
+    }
```
Comment (VinnieJ-2k20) [inline src/robotFactory.js:53]: Here you need to just assign `z` to coords. `x` and `y` are handled in `BaseRobot` constructor.

File: src/robotFactory.js [hunk 5]
```diff
@@ -1,10 +1,84 @@
...
 
-class FlyingRobot {}
+    if (!coords.x) {
+      this.coords = {
+        x: 0,
+        y: 0,
+      };
+    } else {
+      this.coords = coords;
+    }
```
Comment (VinnieJ-2k20) [inline src/robotFactory.js:16]: Learn to use logical operators.
```suggestion
  this.coords = {
    x: coords.x || 0,
    y: coords.y || 0,
  }
```

File: src/robotFactory.js (hunk 2 above)
Comment (VinnieJ-2k20) [reply src/robotFactory.js:73]: Agree with Mykhailo.

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
Almost done, let's just further improve your code.

**Review by VinnieJ-2k20 — APPROVED** [review]
Good job 👍

**Pull request closed without merging**